app/backend/recipes/src/main/java/com/juotava/recipes/service/RecipesService.java

```diff
@@ -272,6 +287,79 @@ private String encodeImageToBase64(BufferedImage image) {
+                throw new Exception();
+            }
+        } catch (Exception e){
+            System.out.println("INFO: No drink of the day for today, creating new ...");
+            List<Recipe> recipes = this.recipeRepository.findAllPublished();
+            if (!recipes.isEmpty()){
+                DrinkOfTheDay drinkOfTheDay = this.generateDrinkOfTheDay(LocalDate.now(), recipes);
+                if (drinkOfTheDay != null){
+                    this.drinkOfTheDayRepository.save(drinkOfTheDay);
+                }
+                return drinkOfTheDay;
+            } else {
+                System.out.println("ERROR: There are no recipes from which a drink of the day could be chosen");
+                return null;
+            }
+        }
+
+    }
+
+    public DrinkOfTheDay generateDrinkOfTheDay(LocalDate date, List<Recipe> recipes){
+        String weather = "";
+        if (date.equals(LocalDate.now())){
+            weather = requestTodaysWeather();
+        }
+
+        String recipesString = recipes.stream().map(Recipe::toRecipeOfTheDayString).collect(Collectors.joining(","));
+        TextRequest request = new TextRequest(textModel);
+        request.addMessage(new Message("system", "You are a deciding machine. I will give you a list of Recipes in the format <ID: XXX, Title: YYY, Description: ZZZ>, a date and some weather data. Based on these information you must decide on which Recipe you choose as the drink of the day. Keep in mind that some drinks are better fitted in certain seasons or temperatures. Consider Holidays or special Events. Give a sound reasoning text in German. Use the colloquial 'Du'. Include some Trivia in the reasoning. Do not mention exact numbers on weather data in the reasoning. Your answer must under all circumstances follow: <ID: XXX, REASON: YYY>. Do not write any additional text!"));
+        request.addMessage(new Message("user",
+                "Today is "+date.toString() +" " +
+                        weather + " " +
+                        "The recipes are: ( "+ recipesString +")"
+        ));
+
+        for (int i = 0; i < 3; i++) {
+            try {
+                TextResponseSuccess response = openaiRestTemplate.postForObject(textApiUrl, request, TextResponseSuccess.class);
+                if (response == null || response.getChoices() == null || response.getChoices().isEmpty()){
+                    return null;
+                }
+                DrinkOfTheDay drinkOfTheDay = new DrinkOfTheDay(
+                        date, response.getChoices().get(0).getMessage().getContent()
+                );
+                drinkOfTheDay.setRecipe(
+                        this.recipeRepository.findByUuid(drinkOfTheDay.getParsedRecipeUuid())
+                );
+                return  drinkOfTheDay;
+            } catch (Exception e){
+                System.out.println("ERROR: Versuch "+(i+1)+". Ein Drink des Tages konnte nicht generiert werden. Vermutlich hatte GPT keine Lust 🥲");
```

Comment:
Is it for all systems possible to display emojis? Otherwise I would, altough it's very funny, leave it out

Reply:
You are right, also it was in german. I changed it